### PR TITLE
Batch product listing (photos) basic flow

### PR DESCRIPTION
**Pull request by marcusvx**
- Upgrades React to v18.x
- Add craco configuration, allowing customization of webpack config
- Add Ant Design to the project
- Add Uppy and Transloadit
- Implement basic flow for batch product listing (photos) 
- Fix some failing unit tests

### CODE REVIEW

File: craco.config.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+const { InjectManifest } = require('workbox-webpack-plugin');
+
+module.exports = {
+  reactScriptsVersion: 'sharetribe-scripts',
+  webpack: {
+    configure: {
+      module: {
+        rules: [
+          {
+            test: /\.m?js$/,
```
Comment (marcusvx) [inline craco.config.js:10]: This is need to make Uppy's Dashboard work.

File: craco.config.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+            resolve: {
+              fullySpecified: false,
+            },
+          },
+        ],
+      },
+    },
+    plugins: [
+      new InjectManifest({
```
Comment (marcusvx) [inline craco.config.js:19]: This is to allow Uppy's Golden Retriever plugin work with large files, see:
https://uppy.io/docs/golden-retriever/#enabling-service-worker

File: src/components/NamedLink/NamedLink.js [hunk 3]
```diff
@@ -58,17 +58,6 @@ export const NamedLinkComponent = props => {
   );
 };
 
-NamedLinkComponent.defaultProps = {
```
Comment (marcusvx) [inline src/components/NamedLink/NamedLink.js:61]: Default props will be removed in future versions of React. I couldn't upgrade React to the latest v18 versions because it added a lot of annoying warnings for each use of `defaultProps` and `propTypes`:
See: https://github.com/facebook/react/pull/16210

File: src/store.js [hunk 4]
```diff
@@ -1,16 +1,22 @@
...
 /**
  * Create a new store with the given initial state. Adds Redux
  * middleware and enhancers.
  */
 export default function configureStore(initialState = {}, sdk = null, analyticsHandlers = []) {
-  const middlewares = [thunk.withExtraArgument(sdk), analytics.createMiddleware(analyticsHandlers)];
+  const middlewares = [
+    thunk.withExtraArgument(sdk),
+    analytics.createMiddleware(analyticsHandlers),
+    uppyReduxStore.middleware(),
```
Comment (marcusvx) [inline src/store.js:18]: Uppy is connected with the main Redux store, so we have only one source of truth, as it's suggested by Redux guidelines.

File: package.json [hunk 5]
```diff
@@ -44,8 +62,9 @@
     "prop-types": "^15.8.1",
     "query-string": "^7.1.1",
     "raf": "^3.4.0",
-    "react": "^17.0.2",
-    "react-dom": "^17.0.2",
+    "react": "18.2.0",
+    "react-dates": "^21.8.0",
```
Comment (rafaerod89) [inline package.json:66]: I think we won't need this anymore right?

File: server/api/transloadit-params.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    .utc()
+    .add(1, 'hour')
+    .format('YYYY/MM/DD HH:mm:ss Z');
+
+  const params = {
+    auth: {
+      key: authKey,
+      expires,
+    },
+    template_id: process.env.TRANSLOADIT_UPLOAD_LISTING_ASSETS_TEMPLATE_ID,
```
Comment (rafaerod89) [inline server/api/transloadit-params.js:23]: Did you add this key and any other required to the Secret Manager?

File: server/scripts/events/notifyProductListingCreated.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    console.error('Error processing event:', error);
+  }
+};
+
+function script() {
+  const integrationSdk = integrationSdkInit();
+  const storageManagerClient = new StorageManagerClient();
+
+  const queryEvents = args => integrationSdk.events.query({ ...args, eventTypes: EVENT_TYPES });
+  const analyzeEvent = event => processEvent(integrationSdk, event, storageManagerClient);
```
Comment (rafaerod89) [inline server/scripts/events/notifyProductListingCreated.js:47]: Ideally this is where you need you need to check if it's a valid event (remember to check the listing type and category). Check how I do it in 'marketplace/server/scripts/events/notifyProfileListingUpdated.js'

File: server/scripts/events/notifyProductListingCreated.js [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+  try {
+    const originalAssetData = await storageManagerClient.uploadOriginalAsset(
+      userId,
+      listingId,
+      imageUrl
+    );
+    const imageStream = await httpFileUrlToStream(imageUrl);
+    const { data: sdkImage } = await integrationSdk.images.upload({ image: imageStream });
+
+    await integrationSdk.listings.update(
```
Comment (rafaerod89) [inline server/scripts/events/notifyProductListingCreated.js:29]: After this we need to communicate with Slack to start the approval flow. I'll be taking care of this, but just wanted to point it out

File: src/styles/antDesignTokens.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
+import './marketplaceDefaults.css';
+
+export const createTheme = branding => {
```
Comment (rafaerod89) [inline src/styles/antDesignTokens.js:3]: Very nice!!

File: src/components/UserNav/UserNav.js [hunk 10]
```diff
@@ -12,20 +12,13 @@ const UserNav = props => {
...
-      },
-    },
-    {
-      text: <FormattedMessage id="UserNav.favoriteListings" />,
-      selected: currentPage === 'FavoriteListingsPage',
-      linkProps: {
-        name: 'FavoriteListingsPage',
-      },
-    },
+    // {
```
Comment (rafaerod89) [inline src/components/UserNav/UserNav.js:15]: Why did you change this file?

File: src/routing/routeConfiguration.js [hunk 11]
```diff
@@ -134,6 +136,24 @@ const routeConfiguration = (layoutConfig, accessControlConfig) => {
       component: EditListingPage,
       loadData: pageDataLoadingAPI.EditListingPage.loadData,
     },
+    {
+      path: '/listing/new',
```
Comment (rafaerod89) [inline src/routing/routeConfiguration.js:140]: Ain't this too general?

Let's discuss this some more, but I think it should be a bit more specific given that is ONLY for produts

File: src/routing/routeConfiguration.js [hunk 12]
```diff
@@ -134,6 +136,24 @@ const routeConfiguration = (layoutConfig, accessControlConfig) => {
...
+      auth: true,
+      component: () => (
+        <NamedRedirect
+          name="BatchEditListingPage"
+          params={{ slug: draftSlug, id: draftId, type: 'new', tab: 'upload' }}
+        />
+      ),
+    },
+    {
+      path: '/listing/:tab',
```
Comment (rafaerod89) [inline src/routing/routeConfiguration.js:151]: Same here. Let's touch base on this route

File: src/containers/BatchEditListingPage/BatchEditListingPage.js [hunk 13]
```diff
@@ -0,0 +1,106 @@
...
+import { hasPermissionToPostListings, isUserAuthorized } from '../../util/userHelpers';
+import { NamedRedirect, Page } from '../../components';
+import TopbarContainer from '../../containers/TopbarContainer/TopbarContainer';
+
+import css from './BatchEditListingPage.module.css';
+import { initializeUppy, requestSaveBatchListings } from './BatchEditListingPage.duck';
+import BatchEditListingWizard from './BatchEditListingWizard/BatchEditListingWizard';
+
+export const BatchEditListingPageComponent = props => {
+  const { currentUser, history, intl, params, page, onInitializeUppy, onSaveBatchListing } = props;
```
Comment (rafaerod89) [inline src/containers/BatchEditListingPage/BatchEditListingPage.js:21]: What about the 'tab' field we pass from the router?

File: src/containers/ListingPage/ActionBarMaybe.js [hunk 14]
```diff
@@ -108,9 +108,6 @@ export const ActionBarMaybe = props => {
       [css.ownListingTextPendingApproval]: isPendingApproval,
     });
 
-    const hasValidType = approvalToPublishOptions?.type && approvalToPublishOptions.type !== 'none';
```
Comment (rafaerod89) [inline src/containers/ListingPage/ActionBarMaybe.js:111]: I think this was a mistake?

File: src/containers/EditListingPage/EditListingWizard/EditListingWizard.js [hunk 15]
```diff
@@ -489,7 +489,7 @@ class EditListingWizard extends Component {
 
     // Redirect user to first tab when encoutering outdated draft listings.
     if (invalidExistingListingType && isNewListingFlow && selectedTab !== tabs[0]) {
-      return <NamedRedirect name="EditListingPage" params={{ ...params, tab: tabs[0] }} />;
```
Comment (rafaerod89) [inline src/containers/EditListingPage/EditListingWizard/EditListingWizard.js:492]: We sure about this? Ain't this file used for the OLD editing flow and not the new one?

File: src/containers/ManageListingsPage/ManageListingCard/ManageListingCard.js [hunk 16]
```diff
@@ -127,7 +127,7 @@ const ShowFinishDraftOverlayMaybe = props => {
       >
         <NamedLink
           className={css.finishListingDraftLink}
-          name="EditListingPage"
```
Comment (rafaerod89) [inline src/containers/ManageListingsPage/ManageListingCard/ManageListingCard.js:130]: Let's talk about this later today. If it's ok to completely overwrite the link or make it conditional

File: src/translations/de.json [hunk 17]
```diff
@@ -33,6 +33,25 @@
   "AuthenticationPage.verifyEmailTitle": "{name}, bitte überprüfe deinen E-Mail-Posteingang, um deine E-Mail-Adresse zu bestätigen",
   "Avatar.bannedUserDisplayName": "Gesperrter Benutzer",
   "Avatar.deletedUserDisplayName": "Gelöschter Benutzer",
+  "BatchEditListingUploaderPanel.title": "Fotos hochladen",
```
Comment (rafaerod89) [inline src/translations/de.json:36]: None of this is needed. But did you add this on the Sharetrive console?

If not let's touch base on how to do it because it is tricky

File: src/translations/es.json [hunk 18]
```diff
@@ -33,6 +33,25 @@
   "AuthenticationPage.verifyEmailTitle": "{name}, revisa tu cuenta de correo para poder verificar tu cuenta.",
   "Avatar.bannedUserDisplayName": "Usuario bloqueado",
   "Avatar.deletedUserDisplayName": "Usuario eliminado",
+  "BatchEditListingUploaderPanel.title": "Subir fotos",
```
Comment (rafaerod89) [inline src/translations/es.json:36]: Same here

File: src/translations/fr.json [hunk 19]
```diff
@@ -33,6 +33,25 @@
   "AuthenticationPage.verifyEmailTitle": "{name}, consultez votre boîte de réception pour confirmer votre adresse email",
   "Avatar.bannedUserDisplayName": "Utilisateur banni",
   "Avatar.deletedUserDisplayName": "Utilisateur supprimé",
+  "BatchEditListingUploaderPanel.title": "Télécharger des photos",
```
Comment (rafaerod89) [inline src/translations/fr.json:36]: Same here

File: src/util/api.js [hunk 20]
```diff
@@ -144,3 +144,13 @@ export const transitionPrivileged = body => {
 export const createUserWithIdp = body => {
   return post('/api/auth/create-user-with-idp', body);
 };
+
+///
```
Comment (rafaerod89) [inline src/util/api.js:148]: I think you forgot to remove this comments :P

File: src/containers/ManageListingsPage/ManageListingsPage.js [hunk 21]
```diff
@@ -165,70 +240,23 @@ export const ManageListingsPageComponent = props => {
...
-                hasClosingError={closingErrorListingId.uuid === l.id.uuid}
-                renderSizes={renderSizes}
-              />
-            ))}
+          <div className={css.listingCardsTabs}>
+            <Tabs
+              defaultActiveKey={currentListingType}
+              onChange={onTabChange}
+              items={[
+                { key: 'product-listing', label: 'Shop', children: listingRenderer },
```
Comment (rafaerod89) [inline src/containers/ManageListingsPage/ManageListingsPage.js:248]: Don't use the types directly as string, instead when possible use this:
LISTING_TYPES
marketplace/src/util/types.js
marketplace/server/api-util/metadataHelper.js
(depending on if you are on the FE or the BE)

File: src/containers/BatchEditListingPage/BatchEditListingPage.js (hunk 13 above)
Comment (marcusvx) [reply src/containers/BatchEditListingPage/BatchEditListingPage.js:21]: It's being used in src/containers/BatchEditListingPage/BatchEditListingWizard/BatchEditListingWizard.js.

BTW, I've updated the route to:
`/listing/:category/:type/:tab`

**Review by rafaerod89 — APPROVED** [review]
